source/service.go

```diff
@@ -379,6 +380,11 @@ func (sc *serviceSource) extractNodeTargets() (endpoint.Targets, error) {
 
 	nodes, err := sc.client.CoreV1().Nodes().List(metav1.ListOptions{})
 	if err != nil {
+		if errors.IsForbidden(err) {
+			// Return an empty list because it makes sense to continue and try other sources.
+			log.Debugf("Unable to list nodes (Forbidden), NodePort services will be skipped")
+			return internalIPs, nil
```

Comment:
Yes, I was a bit unsure about the message. The "NodePort services will be skipped" is a bit more user-centric so that it might be more clear why NodePort services are not getting a DNS entry.

How about `"Unable to list nodes (Forbidden), returning empty list of targets (NodePort services will be skipped)"` or `"Unable to list nodes (Forbidden), returning empty list of targets"`?

Reply:
+1 for the first suggestion, also return endpoints.Targets instead of internalIPs maybe
internalIPs might be confusing to read when return nothing

Reply:
Yes, good point! Fixed! 👍